libgpos/src/memory/CMemoryPoolManager.cpp

```diff
@@ -93,12 +92,16 @@ CMemoryPoolManager::CMemoryPoolManager
 //
 //---------------------------------------------------------------------------
 GPOS_RESULT
-CMemoryPoolManager::EresInit()
+CMemoryPoolManager::EresInit
+	(
+		void* (*pfnAlloc) (SIZE_T),
+		void (*pfnFree) (void*)
+	)
 {
 	GPOS_ASSERT(NULL == CMemoryPoolManager::m_pmpm);
 
 	// raw allocation of memory for internal memory pools
-	void *pvAllocBase = PvMalloc(sizeof(CMemoryPoolMalloc));
+	void *pvAllocBase = PvMalloc(sizeof(CMemoryPoolAlloc));
```

Comment:
We use malloc to allocate memory for the CMemoryPoolAlloc and the CMemoryTracker structures needed to initialize CMemoryPoolManager. This was done at the init time of gpos earlier, which is why this was necessary, but I don't really see why it would be anymore, since we are passed in the allocators anyway.